apps/sentry-client-desktop/src/hooks/useChainDataWithCallback.ts

```diff
@@ -96,42 +104,52 @@ export function useChainDataWithCallback() {
 	}, [pools]);
 
 	// return ownersKycMap & combinedWalletsKycMap
-	useEffect(() => {
-		setChainState((_chainState) => {
-			const ownersKycMap = Object.keys(combinedWalletsKycMap).reduce((result, wallet) => {
-				if (owners.includes(wallet)) {
-					result[wallet] = combinedWalletsKycMap[wallet];
-				}
-				return result;
-			}, {});
-
-			return {
-				..._chainState,
-				ownersKycMap,
-				combinedWalletsKycMap,
-			};
-		});
-	}, [JSON.stringify(combinedWalletsKycMap), owners]);
+	// useEffect(() => {
```

Comment:
I love the way you abstracted this out of here. I know it's style preferences, but I love when you break things up into smaller pieces. Thank you for making it more readable and maintainable!